scripts/compile-tsc.js

```diff
@@ -27,7 +28,7 @@ function getCommand(watch) {
     args.push('-w');
   }
 
-  return `${tsc} ${args.join(' ')}`;
+  return `${tsc} ${args.join(' ')} && ${downlevelDts} dist ts3.5/dist`;
```

Comment:
After trying to add parameters to make this configurable, I figured it was cleaner and more rational to just hardcode it.

The src and target directories for `downlevel-dts` must work in tandem with the `package.json` so passing it through as a parameter doesn't seem to add any value.

Let me know if there's some other flexibility needs I need to account for!

Reply:
Also, for those wondering "Why ts3.5 as the target dir name?" (okay maybe no one is wondering that...)

`downlevel-dts` doesn't target a TS version. It just makes the latest TS (3.8 in this case) compatible with older versions via transformations of a d.ts file. I had to do some digging into what @sandersn was targeting since the [repo](https://github.com/sandersn/downlevel-dts) is strangely quiet about it. The README recommends `ts3.4` but that didn't seem right to me since TS3.5 users would benefit as well.

So I read [Twitter](https://twitter.com/sanders_n/status/1217865246557126656) instead and it looks like it ships types for TS3.5 or earlier. Thus, the naming for the target directory: `ts3.5`.

Reply:
😂

Reply:
I think we should output in `dist/ts3.5` instead of `ts3.5/dist`. Some reasons for this thought:
 - Everything that is "compiled/transpiled" is in a single folder: `dist`
 - It's a common practice to have the result of a compilation stage output into `dist`
 - `dist/**` is listed in `files` attributes of `package.json` (https://github.com/storybookjs/storybook/pull/9847/commits/038b7f9d76ae8de5c324d56dbf7b5ad876ea4dd5#diff-353aed04b1a7ddb0879e79df26422780R29) meaning it is packaged inside NPM package. 
⚠️ If we keep `ts3.5/dist` we must add it to the `files` attributes of all packages.  

@lychyi what is your opinion on that?

Reply:
downlevel-dts started off as a tiny one-off piece of code, so I've been slower than I should to add real documentation. I just added some, so the README has substantially more detail now.